### DISCUSSION TITLE
Integration with sops

### DISCUSSION THREAD

**Discussion by zephyros-dev** [grafana/tanka#608] [Ideas]
Tanka already support importing a json/yaml file, so I wonder if a similar integration can be made with sops encrypted file?

**Comment by lukasmrtvy** [comment]
i would suggest something like:
`tk apply --yaml-vars=<(sops -d secrets.enc.yaml)` 
or from stdin ( that can work for linux, mac, powershell )
`sops -d secrets.enc.yaml | tk apply --yaml-vars - `

**Reply by Duologic** [reply]
Piping it in with top-level arguments seems to work pretty well:
```
[I] ➜ cat environments/default/main.jsonnet
function(secrets) {
  secrets: secrets,
}

[I] ➜ cat secrets.json
{
    "password": "supersecret"
}

[I] ➜ cat secrets.json | tk eval --tla-code "secrets=import '/dev/stdin'" environments/default/main.jsonnet
{
  "secrets": {
    "password": "supersecret"
  }
}
```

**Reply by Duologic** [reply]
`cat secrets.json` can be the `sops -d secrets.enc.json` for example.

**Reply by zephyros-dev** [reply]
The same can be done with yaml file:
```
# cat environments/default/main.jsonnet
function(secret_yaml) {
  secrets: std.native('parseYaml')(secret_yaml),
}
# sops -d secret.sops.yaml
secret: very_secret
# sops -d secret.sops.yaml | tk eval --tla-code "secret_yaml=importstr '/dev/stdin'" environments/default/main.jsonnet
{
  "secrets": [
    {
      "secret": "very_secret"
    }
  ]
}
```

**Comment by zephyros-dev** [comment]
I see that there's some workaround [here](https://github.com/grafana/tanka/issues/353#issuecomment-928203142), but it's a bit unwieldy, and decrypting multiple files would not be easy to automate. sops already have a [Go decrypt package](https://pkg.go.dev/go.mozilla.org/sops/decrypt?utm_source=godoc), so an ideal integration I can think of would be done using native function, something like:
```
{
  someField:  std.native('sopsDecrypt')(importstr 'secret.sops.yaml'),
}
```

**Comment by zephyros-dev** [comment]
I made a sample gist for using tanka with encrypted sops yaml files [here](https://gist.github.com/zephyros-dev/08ab08643e3a58bde6f93e7ba2d9f915). The name of the sops files is hardcoded at the moment however.
